src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+"""Transform raw AEO tables into normalized assets.
+
+Raw AEO tables often contain many different types of data which are split out
+along different dimensions. For example, one table may contain generation
+split out by fuel type as well as prices split out by service category.
+
+As a result, we need to split these large tables into smaller tables that have
+more uniform data, which we do by filtering the large table to its relevant
+subsets, and then transforming some human-readable string fields into useful
+metadata fields.
+"""
+
+import numpy as np
+import pandas as pd
+from dagster import asset
+
+from pudl.helpers import apply_pudl_dtypes
+
+
+def __sanitize_string(series: pd.Series) -> pd.Series:
+    return series.str.lower().str.strip().str.replace(r"\W+", "_", regex=True)
+
+
+def get_series_info(series_name: pd.Series) -> pd.DataFrame:
+    """Break human-readable series name into machine-readable fields.
+
+    The series name contains several comma-separated fields: the variable,
+    the region, the case, and the report year.
+
+    The variable then contains its own colon-separated fields: a general topic,
+    a less general subtopic, and specific variable name. It may also contain a
+    fourth field for a specific dimension such as fuel type.
+    """
+    fields = series_name.str.split(",", expand=True).rename(
+        columns={
+            0: "variable_fields",
+            1: "region_series",
+            2: "case_series",
+            3: "report_year_series",
+        }
+    )
+    variable_fields = fields.variable_fields.str.split(" : ", expand=True).rename(
+        columns={0: "topic", 1: "subtopic", 2: "variable_name", 3: "dimension"}
+    )
+    return fields.join(variable_fields)
```

Comment:
ow this is total personal preference but i don't love that you can't tell `join` to join on the index like you can with `merge`. its just nice when the how of the squish is really explicit